### ISSUE TITLE
Backup failing when using RADOS devices as a back-end

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#601]
Originally reported of Google Code with ID 545.

```
What software version are you running? Please provide the output of "gnt-
cluster --version" and "gnt-cluster version".

# gnt-cluster version
Software version: 2.7.0~rc3
Internode protocol: 2070000
Configuration format: 2070000
OS api version: 20
Export interface: 0

<b>What distribution are you using?</b>

Debian 7

<b>What steps will reproduce the problem?</b>
1.gnt-backup export -f -n $HOSTNAME tuxpat
<b>2.</b>
<b>3.</b>

<b>What is the expected output? What do you see instead?</b>
Wed Jul 31 15:31:23 2013 Creating a snapshot of disk/0 on node XXXX
Wed Jul 31 15:31:23 2013  - WARNING: Could not snapshot disk/0 on node XXXX: Error while executing backend function: Invalid object attribute Disk.unique_id


<b>Please provide any additional information below.</b>
My best guess is that this fails because it is trying to perform an lvm snapshot of a rados device.
The temporary workaround would be to add a --no-snapshot option to avoid producing a snapshot (and forcing shutdown of the machine)

In the long term, Ganeti should understand the type of back-end and perform a rbd snap create instead
 



```
Originally added on 2013-07-31 13:41:29 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Assigning to rsanti@ as he's taking care of shared storage
```
Originally added on 2013-08-27 07:00:46 +0000 UTC.

Added Labels: SmallTask
Added to Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Any news on this? IMHO, not being able to backup an instance would warrant a more critical priority. Did backups for RBD based instances work in 2.9?
```
Originally added on 2014-02-07 09:20:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-02-19 10:06:21 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Move all issues assigned to 2.10 to 2.11, because their priority would not justify an inclusion in 2.10.
```
Originally added on 2014-02-19 10:07:59 +0000 UTC.

Added to Milestone: Release2.11
Removed from Milestone: Release2.10

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi, 
are there any new on that issue?
I need to backup instances on my cluster, is there a workaround if there is no possibility to do it with "gnt-backup"?
```
Originally added on 2014-04-30 08:52:39 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-05-05 08:50:10 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hello,
I'm using a customized ext storage provider and it fails in the very same way:

WARNING: Could not snapshot disk/0 on node node1: Error while executing backend function: Invalid object attribute Disk.unique_id

```
Originally added on 2014-07-10 09:12:44 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Hi,

you are right. Only the BlockDev class has a unique_id. Disk has a logical_id. This should be fixed in stable-2.10. But still backup export will not be functional (currently only plain and drbd support it).

BTW I am working on supporting gnt-backup export for the ext disk template.
I will be submitting a patch set today or tomorrow but it will go on master. So you probably have to backport it yourself. Just for the record which Ganeti version are you using?

Thanks,
dimara
```
Originally added on 2014-07-10 09:45:20 +0000 UTC.

Added to Milestone: Release2.10
Removed from Milestone: Release2.11

**Comment by Ganeti-Issues-Migrator** [comment]
```
I am marking this issue as fixed because it hasn't been updated in a while and there seem to be a couple of patches that might be related to this (like 4ff6552e00da8edce0a9b491d43d744a69839ad5) around the same date as the last comment.

If this is not the case please report so, to help us reduce the effort in migrating all our issues to Github according to: https://groups.google.com/forum/#!topic/ganeti/b2-eukpsXpk
```
Originally added on 2016-12-15 14:44:17 +0000 UTC.

Changed State: Fixed
Removed from Milestone: Release2.10
